pkg/kn/commands/revision/revision_list.go

```diff
@@ -80,6 +83,15 @@ func NewRevisionListCommand(p *commands.KnParams) *cobra.Command {
 					return nil
 				}
 			}
+
+			// sort revisionList by configuration generation key
+			sort.SliceStable(revisionList.Items, func(i, j int) bool {
+				if revisionList.Items[i].Labels[serving.ConfigurationGenerationLabelKey] != revisionList.Items[j].Labels[serving.ConfigurationGenerationLabelKey] {
+					return revisionList.Items[i].Labels[serving.ConfigurationGenerationLabelKey] > revisionList.Items[j].Labels[serving.ConfigurationGenerationLabelKey]
```

Comment:
This is a string comparison, which is wrong. You should first convert to integer (with error check) and then compare.

E.g. ["9" > "10"](https://play.golang.org/p/6MhnASXDT8A)

Reply:
Thanks for pointing it out! (A serious bug here...) I'll fix that.